src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/CodeDirectiveFormattingTest.cs

```diff
@@ -44,6 +44,57 @@ public interface Bar
 ");
         }
 
+        [Theory]
+        [CombinatorialData]
+        public async Task Format_DocumentWithDiagnostics(bool useSourceTextDiffer)
+        {
+            await RunFormattingTestAsync(useSourceTextDiffer: useSourceTextDiffer,
+input: @"
+@page
+@model BlazorApp58.Pages.Index2Model
+@{
+}
+
+<section class=""section"">
+    <div class=""container"">
+        <h1 class=""title"">Managed pohotos</h1>
+        <p class=""subtitle"">@Model.ReferenceNumber</p>
+    </div>
+</section>
+<section class=""section"">
+    <div class=""container"">
+        @foreach       (var item in Model.Images)
+        {
+            <div><div>
+        }
+    </div>
+</section>
+",
+expected: @"@page
+@model BlazorApp58.Pages.Index2Model
+@{
+}
+
+<section class=""section"">
+    <div class=""container"">
+        <h1 class=""title"">Managed pohotos</h1>
+        <p class=""subtitle"">@Model.ReferenceNumber</p>
+    </div>
+</section>
+<section class=""section"">
+    <div class=""container"">
+        @foreach (var item in Model.Images)
+        {
+            <div>
```

Comment:
It looks like formatting destroyed the document? Shouldn't this expect
```
    <div>
    </div>
}
``` and to have `</section>` aligned left?

Reply:
If the second div tag were a close tag, yes, but its an open tag.

Ultimately this comes down to "should we even bother trying to format documents that are in error?". My opinion, probably tainted by C# which does the same, is yes. I think we're probably mostly better off doing as much as we can, and if we truly break things Ctrl+Z is there. I wonder how many "formatting does nothing" reports we've had might have really been caused by having a single diagnostic somewhere in the document, that moved by one space, so we threw away the whole thing.